File: 2586-longest-square-streak-in-an-array/2586-longest-square-streak-in-an-array.py
class Solution:
    def longestSquareStreak(self, nums: List[int]) -> int:
        nums.sort()
        maxlen=0
        map=dict()
        for n in nums:
            sqroot=int(math.sqrt(n))
            if sqroot*sqroot==n and sqroot in map:
                map[n]=map[sqroot]+1
                maxlen=max(maxlen,map[n])
            else:
                map[n]=1
        return maxlen if maxlen>=2 else -1
        

File: 2586-longest-square-streak-in-an-array/2586-longest-square-streak-in-an-array.java
class Solution {
    public int longestSquareStreak(int[] nums) {
        Arrays.sort(nums);
        int maxlen=0,sqrt=0;
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int n:nums){
            sqrt=(int)Math.sqrt(n);
            if(sqrt*sqrt==n && map.containsKey(sqrt)){
                map.put(n,map.get(sqrt)+1);
                maxlen=Math.max(maxlen,map.get(n));
            }
            else{
                map.put(n,1);
            }
        }
        return maxlen>=2?maxlen:-1;
    }
}